gr/include/gnuradio/tag.h

```diff
@@ -32,7 +34,46 @@ class tag_t
     bool operator!=(const tag_t& rhs) const;
 
     void set_offset(uint64_t offset) { _offset = offset; }
-    pmtf::pmt operator[](const std::string& key) const { return _map.at(key); }
+    [[deprecated("unsafe - use at(key) instead")]] pmtf::pmt
+    operator[](const std::string& key) const
+    {
+        return _map.at(key); // implementation violates std::map::operator[] contract
+    }
+
+    pmtf::pmt& at(const std::string& key) { return _map.at(key); }
+    const pmtf::pmt& at(const std::string& key) const { return _map.at(key); }
+
+    [[nodiscard]] std::optional<std::reference_wrapper<const pmtf::pmt>>
+    get(const std::string& key) const noexcept
+    {
+        try { // TODO: missing find function in pmtf::map -> exception based look-up
+            return _map.at(key);
+        } catch (std::out_of_range& e) {
+            return std::nullopt;
+        }
+    }
+    [[nodiscard]] std::optional<std::reference_wrapper<pmtf::pmt>>
+    get(const std::string& key) noexcept
+    {
+        try { // TODO: missing find function in pmtf::map -> exception based look-up
+            return _map.at(key);
+        } catch (std::out_of_range&) {
+            return std::nullopt;
+        }
+    }
+
+    void insert_or_assign(const std::pair<std::string, pmtf::pmt>& value)
+    {
+        // TODO: expose std::map compatible insert & insert_or_assign interface
+        _map[value.first] = value.second;
+    }
+    void insert_or_assign(const std::string& key, const pmtf::pmt& value)
+    {
+        // TODO: expose std::map compatible insert & insert_or_assign interface
+        _map[key] = value;
+        // return _map.insert_or_assign(key, std::move(value));
+    }
```

Comment:
We are in the process of rewriting the pmts to use variants instead of flatbuffers.  It hasn't been integrated into GR4.0 yet.  But you will have direct access to an std::map, rather than a wrapper class.

Reply:
@jsallay thanks for the info. Regarding this specific PR, should I use a different API member method access?